processing/src/main/java/org/apache/druid/segment/incremental/OnheapIncrementalIndex.java

```diff
@@ -1038,4 +1176,316 @@ public void clear()
       facts.clear();
     }
   }
+
+  public static class OnHeapAggregateProjection implements IncrementalIndexRowSelector
```

Comment:
Given this is a static class, and the file is already quite large, please consider making this into its own file.